app/controllers/rentals_controller.rb

```diff
@@ -0,0 +1,67 @@
+class RentalsController < ApplicationController
+  def create
+    rental_info = rental_params
+    rental_info[:title] = params[:title]
+    rental = Rental.new(rental_info)
+    if rental.save
+      render status: :ok, json: {id: rental.id}
+    else
+      render status: :bad_request, json: { errors: rental.errors.messages }
+    end
+  end
+
+  def update
+    movie = Movie.find_by(title: params[:title] )
+    rental = Rental.find_by(movie_id: movie.id, customer_id: rental_params[:customer_id], returned: false)
+    if rental
+      rental.returned = true
+      if rental.save
+        render status: :ok, json: { movie_returned: true }
+      end
+    elsif Rental.find_by(movie_id: movie.id, customer_id: rental_params[:customer_id], returned: true)
+      render status: :bad_request, json: { errors: { check_in: "Customer has already checked in this movie"}}
+    else
+      render status: :bad_request, json: { errors: { check_in: "Cannot find this rental"} }
+    end
+  end
+
+  def overdue
+    #what is the syntax for a conditional for due date?
+    rentals = Rental.where("returned = ? AND due_date <= ?", false, Date.current)
+    if rentals
+      #array of customer ids for customers who have overdue rentals
+      customer_ids = rentals.map { |rental| rental.customer_id }
+
+      overdue_rentals = []
+      customer_ids.each do |id|
+        customer = Customer.find_by_id(id)
+        hash = Hash.new
```

Comment:
For consistency you could do {} (since you do [] for a new array)...but this is me completely nitpicking bc you did such a great job 😁